r/R/arrow-package.R

```diff
@@ -196,6 +196,20 @@ configure_tzdb <- function() {
         )
       )
     }
+
+    # On MacOS only, Check if we are running in under emulation, and warn this will not work
+    rosetta <- identical(sysname, "darwin") && identical(system("sysctl -n sysctl.proc_translated", intern = TRUE), "1")
```

Comment:
I see this is also in install_arrow(), NBD but we could factor this out and put it in arrow-package.R alongside `on_macos_10_13_or_lower()` et al.